docs/docs/tutorials/todos.md

```diff
@@ -0,0 +1,665 @@
+  Future<Todo> create(Todo todo);
+
+  /// Return all todos.
+  Future<List<Todo>> readAll();
+
+  /// Return a todo with the provided [id] if one exists.
+  Future<Todo?> read(String id);
+
+  /// Update the todo with the provided [id] to match [todo] and
+  /// return the updated todo.
+  Future<Todo> update(String id, Todo todo);
+
+  /// Delete the todo with the provided [id] if one exists.
+  Future<void> delete(String id);
+}
+```
+
+We're done with the `todos_data_source`! Next, we'll create a concrete implementation of the `TodosDataSource` interface which is backed by an in-memory cache.
+
+## In-Memory Todos Data Source
+
+Just like with the `todos_data_source`, we'll create a new package called `in_memory_todos_data_source` to contain the concrete implementation.
+
+### Creating `package:in_memory_todos_data_source`
+
+From the root of the project we can use `mason make` to generate a new Dart package again:
+
+```bash
+mason make very_good_dart_package --project_name "in_memory_todos_data_source" --description "An in-memory implementation of the TodosDataSource interface." -o packages
+```
+
+### Updating the `pubspec.yaml`
+
+Next, let's update the `pubspec.yaml` in the `in_memory_todos_data_source` to include the relevant dependencies:
+
+```yaml
+name: in_memory_todos_data_source
+description: An in-memory implementation of the TodosDataSource interface.
+version: 0.1.0+1
+publish_to: none
+
+environment:
+  sdk: '>=2.17.0 <3.0.0'
+
+dependencies:
+  todos_data_source:
+    path: ../todos_data_source
+  uuid: ^3.0.6
+
+dev_dependencies:
+  mocktail: ^0.3.0
+  test: ^1.19.2
+  very_good_analysis: ^3.0.1
+```
+
+:::note
+The `in_memory_todos_data_source` depends on the `todos_data_source` via `path`.
+:::
+
+Install the newly added dependencies via:
+
+```bash
+dart pub get
+```
+
+### Creating the `InMemoryTodosDataSource`
+
+Next, let's update `lib/src/in_memory_todos_data_source.dart` to implement the `TodosDataSource` interface:
+
+```dart
+import 'package:todos_data_source/todos_data_source.dart';
+import 'package:uuid/uuid.dart';
+
+/// An in-memory implementation of the [TodosDataSource] interface.
+class InMemoryTodosDataSource implements TodosDataSource {
+  /// Map of ID -> Todo
+  final _cache = <String, Todo>{};
+
+  @override
+  Future<Todo> create(Todo todo) async {
+    final id = const Uuid().v4();
+    final createdTodo = todo.copyWith(id: id);
+    _cache[id] = createdTodo;
+    return createdTodo;
+  }
+
+  @override
+  Future<List<Todo>> readAll() async => _cache.values.toList();
+
+  @override
+  Future<Todo?> read(String id) async => _cache[id];
+
+  @override
+  Future<Todo> update(String id, Todo todo) async {
+    return _cache.update(id, (value) => todo);
+  }
+
+  @override
+  Future<void> delete(String id) async => _cache.remove(id);
+}
+```
+
+That's it! We're done with the data sources for our Dart Frog application and we're ready to start working on the Dart Frog app itself!
```

Comment:
```suggestion
That's it! We're done making the data sources for our Dart Frog application and we're ready to start working on the Dart Frog app itself!
```
I feel like we needed a verb here so I added "making" but feel free to switch out with another